addons/mail/static/src/models/composer/composer.js

```diff
@@ -19,18 +19,66 @@ function factory(dependencies) {
         // Public
         //----------------------------------------------------------------------
 
-        closeMentionSuggestions() {
+        closeSuggestions() {
+            if (this.activeElementName) {
+                this.update({
+                    [this.activeElementName]: [['unlink']],
+                });
+            }
+            if (this.extraListName) {
+                this.update({
+                    [this.extraListName]: [['unlink-all']],
+                });
+            }
+            if (this.mainListName) {
+                this.update({
+                    [this.mainListName]: [['unlink-all']],
+                });
+            }
             this.update({
-                activeSuggestedPartner: [['unlink']],
-                extraSuggestedPartners: [['unlink-all']],
-                mainSuggestedPartners: [['unlink-all']],
+                activeElementName: "",
+                delimiter: "",
+                extraListName: "",
+                mainListName: "",
             });
         }
 
         /**
          * Hides the composer, which only makes sense if the composer is
          * currently used as a Discuss Inbox reply composer.
          */
+        detectDelimiter() {
+            const delimiter = this.textInputContent.substring(this.textInputCursorStart - 1, this.textInputCursorStart);
+            const delimiters = ['@', ':', '#', '/'];
+            if(delimiters.includes(delimiter) && !this.hasSuggestions) {
+                this.update({ delimiter: delimiter });
```

Comment:
Maybe `this.delimiter` should be computed from `this.activeElementName` (or the other way around, not sure) instead of being both written separately, due to all the risk that implies in regard to its synchronization (issue when writing one without the other, ...).

Then in chain all the others (`elementModel`, `mainListName`, `extraListName`, ...) can be computed too. This would make the logic of the current method generic instead of specific to the flux on which this method is called, and this would ensure that no inconsistent state can be written.